.gitignore

```diff
@@ -30,3 +30,32 @@
 *.exe
 *.out
 *.app
+
+# Build Files
+**/CMakeFiles/
+**/CMakeCache.txt
+**cmake_install.cmake
+compile_commands.json
+Makefile
+**CTestTestfile.cmake
+test/**.cmake
+/cmake-build-debug/
+
+# Annoying OS Generated Files
+.DS_Store
+.DS_Store?
+._*
+.Spotlight-V100
+.Trashes
+ehthumbs.db
+Thumbs.db
+.directory
+*~
+
+# IDE files
+.vs/
+.idea/
+.vscode/
+language.settings.xml
+**/**.swp
+
```

Comment:
We can keep these OS- and IDE-specific files as they are fairly common and not caused by user error.

Reply:
Do you want these files pushed to the repo?

Reply:
There are N number of setups, are you going to ignore files for all of them?  
These files are trivial to spot in PRs, the submitter can just be asked to remove them.

Reply:
@RalphSteinhagen Huh? No, I meant that we can keep the ignores for them.

Reply:
I agree we couldn't possibly account for every setup. But there is some value in having these ones in by default. Especially the Vim swap and backup files, as they do really pollute the git status output.